### PR TITLE
Add Ingress support

### PR DESCRIPTION
**Pull request by arturdzm**
**What this PR does / why we need it?**:

- Uses  Ingress kind when Route is not available
- Allows to use expose flag and route fields with ingress  controllers

### CODE REVIEW

File: pkg/controller/runtimecomponent/runtimecomponent_controller.go [hunk 1]
```diff
@@ -599,6 +601,25 @@ func (r *ReconcileRuntimeComponent) Reconcile(request reconcile.Request) (reconc
...
+				appstacksutils.CustomizeIngress(ing, instance)
+				return nil
+			})
+			if err != nil {
+				reqLogger.Error(err, "Failed to reconcile Ingress")
+				return r.ManageError(err, common.StatusConditionTypeReconciled, instance)
+			}
+		} else {
+			ing := &networkingv1beta1.Ingress{ObjectMeta: defaultMeta}
+			err = r.DeleteResource(ing)
```
Comment (navidsh) [inline pkg/controller/runtimecomponent/runtimecomponent_controller.go:616]: For the Knative scenario, please delete the `ingress` resource

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -823,3 +825,55 @@ func GetAppContainer(containerList []corev1.Container) *corev1.Container {
 	}
 	return &containerList[0]
 }
+
+// CustomizeIngress customizes ingress resource
+func CustomizeIngress(ing *networkingv1beta1.Ingress, ba common.BaseComponent) {
+	obj := ba.(metav1.Object)
+	ing.Labels = ba.GetLabels()
+	ing.Annotations = MergeMaps(ing.Annotations, ba.GetAnnotations())
```
Comment (navidsh) [inline pkg/utils/utils.go:833]: Suggestion: you can merge these two lines as follows:
```go
ing.Annotations = MergeMaps(ing.Annotations, ba.GetAnnotations(), ba.GetRoute().GetAnnotations())
```

File: pkg/controller/runtimecomponent/runtimecomponent_controller.go (hunk 1 above)
Comment (arturdzm) [reply pkg/controller/runtimecomponent/runtimecomponent_controller.go:616]: Good catch

File: pkg/controller/runtimecomponent/runtimecomponent_controller.go [hunk 3]
```diff
@@ -599,6 +601,25 @@ func (r *ReconcileRuntimeComponent) Reconcile(request reconcile.Request) (reconc
 		}
 	} else {
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", routev1.SchemeGroupVersion.String()))
+		if instance.Spec.Expose != nil && *instance.Spec.Expose {
+			ing := &networkingv1beta1.Ingress{ObjectMeta: defaultMeta}
```
[the remark this answers is not in the document]
Comment (arturdzm) [reply pkg/controller/runtimecomponent/runtimecomponent_controller.go:605]: Yes, its part of core k8s

**Comment by navidsh** [comment]
Agreed to fix the Knative issue in a separate PR

**Pull request merged**